platform-emf/src/test/java/com/softicar/platform/emf/attribute/field/enums/table/row/EmfEnumTableRowInputEngineTest.java

```diff
@@ -0,0 +1,80 @@
+package com.softicar.platform.emf.attribute.field.enums.table.row;
+
+import com.softicar.platform.common.core.i18n.LanguageEnum;
+import com.softicar.platform.common.core.locale.CurrentLocale;
+import com.softicar.platform.common.core.locale.Locale;
+import com.softicar.platform.db.runtime.test.AbstractDbTest;
+import java.util.Collection;
+import java.util.stream.Collectors;
+import org.junit.Test;
+
+public class EmfEnumTableRowInputEngineTest extends AbstractDbTest {
+
+	private final EmfEnumTableRowInputEngine<EmfTestEnumRow> inputEngine;
+
+	public EmfEnumTableRowInputEngineTest() {
+
+		this.inputEngine = new EmfEnumTableRowInputEngine<>(EmfTestEnumRow.TABLE);
+	}
+
+	@Test
+	public void testWithEmptyPattern() {
+
+		assertEquals("[]", toDisplayStrings(inputEngine.findMatches("", 0)));
+		assertEquals("[Five]", toDisplayStrings(inputEngine.findMatches("", 1)));
+		assertEquals("[Five, Four, One]", toDisplayStrings(inputEngine.findMatches("", 3)));
+		assertEquals("[Five, Four, One, Three, Two]", toDisplayStrings(inputEngine.findMatches("", 5)));
+		assertEquals("[Five, Four, One, Three, Two]", toDisplayStrings(inputEngine.findMatches("", 10)));
+	}
+
+	@Test
+	public void testWithOneLetterPattern() {
```

Comment:
We should probably use "e" instead of "f" in this test case... just to ensure that the implementation does _not only_ check the first character.

Reply:
did so